outlines/fsm/fsm.py

```diff
@@ -114,27 +115,48 @@ def __init__(
         regex_string: str,
         tokenizer: "Tokenizer",
     ):
-        regex_pattern = interegular.parse_pattern(regex_string)
-        regex_fsm, _ = make_deterministic_fsm(regex_pattern.to_fsm().reduce())
+        def func_cache_key_args(
+            regex_string: str, tokenizer: "Tokenizer"
+        ) -> Tuple[str, tuple]:
+            """Return the values that will be used to create the cache key of create_states_mapping"""
+            cacheable_vocabulary = tuple(sorted(tokenizer.vocabulary.items()))
+            return (regex_string, cacheable_vocabulary)
```

Comment:
Why not call the decorator directly with the parameters that are used for the cache key? It would be more explicit.